anesthetic/plot.py

```diff
@@ -59,6 +102,135 @@ def _constructor(self):
+            ax.set_ylabel(labels[y])
+
+        for x, ax in self.ffill(axis=0).iloc[-1, :].dropna().iteritems():
+            ax.set_xlabel(labels[x])
+
+        # left and right ticks and labels
+        for y, ax in self.iterrows():
+            ax_ = ax.dropna()
+            if len(ax_) and ticks == 'inner':
+                for i, a in enumerate(ax_):
+                    if i == 0:  # first column
+                        if a.position == 'diagonal' and len(ax_) == 1:
+                            a.tick_params('y', left=False, labelleft=False)
+                        else:
+                            a.tick_params('y', left=True, labelleft=True)
+                    elif a.position == 'diagonal':  # not first column
+                        tl = a.yaxis.majorTicks[0].tick1line.get_markersize()
+                        a.tick_params('y', direction='out', length=tl/2,
+                                      left=True, labelleft=False)
+                    else:  # not diagonal and not first column
+                        a.tick_params('y', direction='inout',
+                                      left=True, labelleft=False)
+            elif len(ax_) and ticks == 'outer':  # no inner ticks
+                for a in ax_[1:]:
+                    a.tick_params('y', left=False, labelleft=False)
+            elif len(ax_) and ticks is None:  # no ticks at all
+                for a in ax_:
+                    a.tick_params('y', left=False, right=False,
+                                  labelleft=False, labelright=False)
+            else:
+                raise ValueError(
+                    "ticks=%s was requested, but ticks can only be one of "
+                    "['outer', 'inner', None]." % ticks)
+
+        # bottom and top ticks and labels
+        for x, ax in self.iteritems():
+            ax_ = ax.dropna()
+            if len(ax_):
+                if ticks == 'inner':
+                    for i, a in enumerate(ax_):
+                        if i == len(ax_) - 1:  # bottom row
+                            a.tick_params('x', bottom=True, labelbottom=True)
+                        else:  # not bottom row
+                            a.tick_params('x', direction='inout',
+                                          bottom=True, labelbottom=False)
+                            if a.position == 'diagonal':
+                                a.twin.tick_params('x', direction='inout',
+                                                   bottom=True,
+                                                   labelbottom=False)
+                elif ticks == 'outer':  # no inner ticks
+                    for a in ax_[:-1]:
+                        a.tick_params('x', bottom=False, labelbottom=False)
+                elif ticks is None:  # no ticks at all
+                    for a in ax_:
+                        a.tick_params('x', bottom=False, top=False,
+                                      labelbottom=False, labeltop=False)
```

Comment:
Is there any way to do this in fewer lines/a single loop?

Reply:
Probably. However, I don't think it will reduce much in terms of number of lines, and on the downside I think it will become less readable. I think it's cleaner to handle vertical and horizontal axes in two separate loops. They look like duplicates, but there are different kwargs such as `left` and `right` vs `top` and `bottom`, and more importantly there are some subtleties when it comes to the handling of the diagonal axes...

Reply:
It is in its own `_tick_params` method now, though.